src/doxygen.cpp

```diff
@@ -1389,7 +1389,8 @@ static void processTagLessClasses(ClassDef *rootCd,
             if (type.find(icd->name())!=-1) // matching tag less struct/union
             {
               QCString name = md->name();
-              if (md->isAnonymous()) name = "__unnamed__";
+			  /* DGA fix #7556 ANSI-C anonymous (unnamed) struct/unions have duplicated names (__unnamed__) */
+              if (md->isAnonymous()) name = "__unnamed__" + name.right(name.length()-1); /* DGA: ensure unique name */
```

Comment:
why the `name.right(name.length()-1)`, i.e. removing the first character?

Reply:
Because the internal name for anonymous members is first set to something like "@19".
Then it is changed to "__unnamed__" all the time, leading to duplicated members name.
My proposal is to add the internal reference number, but not the "@" character which is not allowed in the identifiers naming.
Hence the removal of the first character to finally have a member name = "__unnnamed__19"